crates/bevy_app/src/app.rs

```diff
@@ -257,6 +263,35 @@ impl App {
             sub_app.run();
         }
 
+        // move schedules out of the world so we can modify it as we iterate
+        // through the schedule events
+        let mut schedules = self
+            .world
+            .remove_resource::<Schedules>()
+            .expect("No Schedules resource in world");
+
+        if let Some(mut sched_events) = self.world.get_resource_mut::<Events<ScheduleEvent>>() {
+            use ScheduleEvent::*;
+
+            for event in self.schedule_event_reader.iter(&sched_events) {
+                match event {
+                    EnableStepping(label)
+                    | DisableStepping(label)
+                    | ToggleStepping(label)
+                    | StepFrame(label)
+                    | StepSystem(label) => {
+                        if let Some(sched) = schedules.get_mut(&**label) {
+                            sched.handle_event(event);
+                        } else {
+                            panic!("unknown Schedule label: #{:?}", label);
```

Comment:
Ditto about error messages: we can just unwrap and get something just as nice.

Reply:
Will do.  I'm not familiar enough with bevy/rust error messages to realize they give helpful information.

Reply:
Yep, as long as you give them good names and store the relevant data in them they format quite nice :)